lib/hexpm_web/templates/dashboard/organization/index.html.eex

```diff
@@ -6,6 +6,92 @@
   </div>
 
   <div class="col-sm-9 organization members">
+    <div class="panel panel-default">
+      <div class="panel-heading">Public profile</div>
+      <div class="panel-body">
+        <div class="form-group">
+          <label>Profile picture</label>
+          <img src="<%= gravatar_url(User.email(@organization.user, :gravatar), :large) %>">
+          <p>
+            <small>
+              Gravatar is used to display organization profile picture. You can update the Gravatar email below or
+              go to <a href="https://en.gravatar.com/emails/">Gravatar</a> to change the image.
+            </small>
+          </p>
+        </div>
+
+        <%= form_for @changeset, Routes.organization_path(Endpoint, :update_profile, @organization), [method: :post, as: :profile], fn f -> %>
+          <div class="form-group">
+            <%= label f, :full_name %>
+            <%= text_input f, :full_name %>
+            <%= error_tag f, :full_name %>
+          </div>
+
+          <div class="form-group">
+            <%= label f, :public_email %>
+            <%= text_input f, :public_email, placeholder: "Your organization's public email", value: @public_email %>
```

Comment:
`value` needs to be set here since `public_email` is not casted in `User.update_profile/2`

Maybe we can add a `cast_assoc` call in `User.update_profile/2` to avoid setting `value` here and also avoid the complex `insert_or_update_or_delete_email_multi` (which basically is a handmade `cast_assoc`)?

Reply:
Yes, if you can make it fit with `cast_assoc` then that seems better.

Reply:
Unfortunately, `cast_assoc` doesn't seem to work well with `Ecto.Multi`. I don't know how to insert `email.add`, `email.public` audit actions if we switch to `cast_assoc`. Because this `user` changeset would change both `user` and `emails`, and it's hard to identify which `emails` were touched in a later `Multi.run`, right?

Off topic: I'm a bit curious if there's anything we can do to make `cast_assoc` work better with `Multi`? Maybe split association changes in a separate `Multi` when called with `Multi.insert` or `Multi.update`?

Reply:
You can store the emails before they changed and retrieve the updated emails from the Multi changes map. You can then compare the old and new and insert the appropriate audit logs. Hard to say if that will be an improvement over the current approach though.

Reply:
> Hard to say if that will be an improvement over the current approach though.

indeed, so I'll just keep the current implementation for now.